launchable/test_runners/bazel.py

```diff
@@ -56,6 +58,35 @@ def f(case: TestCase, suite: TestSuite, report_file: str) -> TestPath:
 
     client.path_builder = f
     client.check_timestamp = False
-    client.scan(base, '**/test.xml')
+
+    if build_event_json:
+        lists = parse_build_event_json(build_event_json)
+        for l in lists:
+            path = Path(base).joinpath(l).resolve()
+            client.scan(path, "**/test.xml")
+    else:
+        client.scan(base, '**/test.xml')
 
     client.run()
+
+
+def parse_build_event_json(path: str) -> List[str]:
+    build_list: List[str] = []
+    try:
+        f = open(path)
+    except Exception as e:
+        click.echo(click.style(
+            "Can not open build event json file {}".format(path), "yellow"), err=True)
+        return build_list
+
+    for line in f:
+        d = json.loads(line)
+        if "id" in d:
+            if "testResult" in d["id"]:
+                if "label" in d["id"]["testResult"]:
+                    label = d["id"]["testResult"]["label"]
+                    # replace //foo/bar:zot to /foo/bar/zot
+                    label = label.lstrip("/").replace(":", "/")
+                    build_list.append(label)
```

Comment:
I was gonna say using a generator would be more Pythonesque but since the whole JSON is parsed eagerly I guess there's no point.

(I do wonder how big this JSON file gets and if we should be parsing this in streaming but we can take that on in the future)

Reply:
thank you for your advice, I changed to use Generator